docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/send-message.md

```diff
@@ -0,0 +1,157 @@
+| `smtp.to`         | `string`  | N        | The email address to send the message to. If omitted, the default recipient configured in the SMTP channel configuration will be used.                                                                                                                               |
+| `smtp.subject`    | `string`  | N        | The subject line for the email. If omitted, a default subject will be used.                                                                                                                                                                                          |
+| `smtp.html`       | `boolean` | N        | Whether the email body should be sent as HTML. If `true`, the message will be treated as HTML content. If `false` or omitted, the message will be treated as plain text.                                                                                             |
+
+### Sending richly formatted messages
+
+By default, the `send-message` step will use the given message and send a plain formatting message
+to the configured channel. However, in some cases you may want to send a richly formatted message,
+such as a Slack message with blocks. To do this, you can use the `encodingType` field to specify the
+format of the message. The supported structures for each message type can be found in the
+[Notification documentation](../90-events/100-notifications/20-message-formatting.md). See the
+example section below for an example of sending a richly formatted Slack message.
+
+:::warning
+
+If `encodingType` is set, all configuration options for the specific channel (e.g. `slack` or
+`smtp`) must be set in the message body itself according to the body format. Any options that are
+set will be ignored. For example, if you want to override the Slack channel ID, you must set it in
+the message body and not in the `slack.channelID` field.
+
+:::
+
+Please note that if you use `json` encoding as a message format, you must ensure that the JSON is
+properly quoted. The step runner assumes that any JSON content should be decoded, so unquoted JSON
+will result in an error. This can be done like so:
+
+```yaml
+steps:
+- uses: send-message
+  config:
+    channel:
+      kind: MessageChannel
+      name: smtp
+    encodingType: json
+    message: |
+      ${{ quote({
+        "subject": "🚀 Deployment Promotion Started",
+        "body": "Kargo has kicked off promotion to stage: " + ctx.stage,
+        "to": [
+          "email@example.com"
+        ],
+        "html": false
+      }) }}
+```
+
+## Examples
+
+### Sending a plain text message to Slack
+
+```yaml
+steps:
+- uses: send-message
+  config:
+    channel:
+      kind: MessageChannel
+      name: slack
+    message: "Kargo has kicked off promotion to stage: ${{ ctx.stage }}."
```

Comment:
I think this might be a more helpful example:

```suggestion
    - if: ${{ always() }}
      uses: send-message
      config:
        channel:
          kind: MessageChannel
          name: slack
        message: "Kargo promotion to ${{ ctx.stage }}: ${{ success() ? 'succeeded' :
          'failed' }}"
```